### ISSUE TITLE
`-inf` appearing in GPT-OSS FSDP training

### ISSUE THREAD

**Issue by pglusacTT** [tenstorrent/tt-blacksmith#542]
### Describe the bug

During training, after ~30 training steps (which show correct loss curve) I see negative infinity in the loss
```
INFO:tt-gpt-oss-20b-test:Current loss and step: 3.2083334922790527 6
Training:   0%|          | 25/67349 [25:07<617:48:00, 33.04s/it]2026-04-22 14:29:36 | INFO | Current loss and step: 3.375 6
INFO:tt-gpt-oss-20b-test:Current loss and step: 3.375 6
Training:   0%|          | 26/67349 [25:28<554:54:18, 29.67s/it]2026-04-22 14:29:58 | INFO | Current loss and step: 3.1666667461395264 6
INFO:tt-gpt-oss-20b-test:Current loss and step: 3.1666667461395264 6
Training:   0%|          | 27/67349 [25:50<510:49:20, 27.32s/it]2026-04-22 14:30:19 | INFO | Current loss and step: 3.7916667461395264 6
INFO:tt-gpt-oss-20b-test:Current loss and step: 3.7916667461395264 6
2026-04-22 14:30:19 | INFO | Step 7 | train/loss: 3.3854
INFO:tt-gpt-oss-20b-test:Step 7 | train/loss: 3.3854
Training:   0%|          | 28/67349 [26:40<639:20:28, 34.19s/it]2026-04-22 14:31:16 | INFO | Current loss and step: -inf 7
INFO:tt-gpt-oss-20b-test:Current loss and step: -inf 7
Training:   0%|          | 29/67349 [27:09<607:07:50, 32.47s/it]2026-04-22 14:31:38 | INFO | Current loss and step: 3.0833334922790527 7
INFO:tt-gpt-oss-20b-test:Current loss and step: 3.0833334922790527 7
Training:   0%|          | 30/67349 [27:31<547:26:31, 29.28s/it]2026-04-22 14:32:00 | INFO | Current loss and step: 2.375 7
INFO:tt-gpt-oss-20b-test:Current loss and step: 2.375 7
Training:   0%|          | 31/67349 [27:53<506:03:45, 27.06s/it]2026-04-22 14:32:22 | INFO | Current loss and step: 2.125 7
INFO:tt-gpt-oss-20b-test:Current loss and step: 2.125 7
2026-04-22 14:32:22 | INFO | Step 8 | train/loss: -inf
```

Might be related https://github.com/tenstorrent/tt-xla/issues/1555.

### Steps to reproduce the issue

Run:
```bash
python3 tt-blacksmith/blacksmith/experiments/torch/gpt_oss/train.py --config tt-blacksmith/blacksmith/experiments/torch/gpt_oss/lora/loudbox/fsdp_gpt_oss_20b_sst2.yaml
```
